util/src/main/java/io/kubernetes/client/util/Config.java

```diff
@@ -32,4 +56,110 @@ public static ApiClient fromCluster() throws IOException {
 
         return result;
     }
+
+    public static ApiClient fromConfig(String fileName) throws IOException {
+        return fromConfig(new FileReader(fileName));
+    }
+
+    public static ApiClient fromConfig(InputStream stream) {
+        return fromConfig(new InputStreamReader(stream));
+    }
+
+    public static ApiClient fromConfig(Reader input) {
+        // Note to the reader: I considered creating a Config object
+        // and parsing into that instead of using Maps, but honestly
+        // this seemed cleaner than a bunch of boilerplate classes
```

Comment:
I think this method getting big because it is trying to do all loading by itself. What about breaking it down like what we did in python client? https://github.com/kubernetes-incubator/client-python/blob/master/kubernetes/base/config/kube_config.py#L108